### PR TITLE
Code related to validating auth

### PR DESCRIPTION
**Pull request by rishipisipati**
#### Required
- [x] I ran the CI build script to ensure that the code passes the following
    - Unit tests
    - Integration tests
    - Linting rules
- [x] I ran the E2E test suite locally to ensure that no regressions were introduced

#### Optional
- [x] My PR contains a DB schema change
    - Reviewers pay special attention to queries/indices when this box is checked
- [ ] My PR is dependent on a change to our infrastructure
    - New SSM keys
    - IAM role updates
    - ETC
- [x] My pull request is fewer than 400 lines of code

## Purpose

* **What is the context of this pull request? Why is it being done?** **required*
Code for implementing authorization safeguards against various user actions.
 
* **Are there any related PRs? Link to them here.**

* **Link to any additional documentation (Jira, Google Docs, wireframes, etc.)** **required when available*
https://betterlesson.atlassian.net/browse/CAE-52

## Changes

* **How are different categories of users affected by these changes?**

* **Are there any changes that need to be called out as significant or particularly tricky?**

* **What is the recommended reading order for these changes? Where should the reviewer start?**

* **Add screenshots here for any UI changes (try to limit to the most important changes to avoid extending the page too much).**

## Release

* **Are there any special considerations for deployment?** **required*

### CODE REVIEW

File: src/main/java/com/betterlesson/learningwalk/learningwalk/controllers/LearningWalkController.java [hunk 1]
```diff
@@ -79,4 +81,52 @@ public LearningWalkClassroom saveDemographicData(
...
+    if (!jwtDecodeService.checkJwtCredentials(auth)) {
+      throw new InvalidTokenException();
+    }
+
+    if (!learningWalkService.isUserAMemberOfLearningWalk(
+        externalId, jwtDecodeService.getUserSubFromJwt(auth))) {
+      throw new UserUnauthorizedForActionException(
+          Constants.LearningWalkActions.VIEW_CLASSROOM_SUMMARY);
+    }
+  }
```
Comment (nicklundin-bl) [inline src/main/java/com/betterlesson/learningwalk/learningwalk/controllers/LearningWalkController.java:115]: Are these test endpoints supposed to be checked in?

Comment (rishipisipati) [reply src/main/java/com/betterlesson/learningwalk/learningwalk/controllers/LearningWalkController.java:115]: Nope, they exist only for the purposes of testing and can be deprecated whenever we want.

File: src/main/java/com/betterlesson/learningwalk/learningwalk/services/LearningWalkService.java [hunk 2]
```diff
@@ -30,6 +30,23 @@ public class LearningWalkService {
...
+    return learningWalkParticipantsRepository
+        .findByLearningWalkIdAndUserSub(learningWalk.getId(), userSub)
+        .isPresent();
+  }
+
+  public Boolean isUserAuthorizedToViewTotalSummary(
+      UUID learningWalkExtId, UUID userSub, String auth) {
+    return (isUserAMemberOfLearningWalk(learningWalkExtId, userSub)
+        || labService.isLabUserManager(auth));
+  }
```
[the remark this answers is not in the document]
Comment (rishipisipati) [reply src/main/java/com/betterlesson/learningwalk/learningwalk/services/LearningWalkService.java:48]: I don't know what you mean.

[the remark this answers is not in the document]
Comment (nicklundin-bl) [reply src/main/java/com/betterlesson/learningwalk/learningwalk/services/LearningWalkService.java:48]: This is good code. Just calling it out

**Review by nicklundin-bl — APPROVED** [review]

**Pull request merged**